egs/chime4/s5_1ch/local/rnnlm/run_lstm_back.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# Copyright 2012  Johns Hopkins University (author: Daniel Povey)
+#           2015  Guoguo Chen
+#           2017  Hainan Xu
+#           2017  Szu-Jui Chen
+
+# This script trains LMs on the reversed Chime4 data, which we
+# call it backward model.
+
+# Begin configuration section.
+affix=1a
+dir=exp/rnnlm_lstm_${affix}_back
+embedding_dim=2048
+lstm_rpd=512
+lstm_nrpd=512
+stage=-10
+train_stage=-10
+
+# variables for lattice rescoring
+ngram_order=4 # approximate the lattice-rescoring by limiting the max-ngram-order
+              # if it's set, it merges histories in the lattice if they share
+              # the same ngram history and this prevents the lattice from 
+              # exploding exponentially
+
+. cmd.sh
+. utils/parse_options.sh
+
+srcdir=data/local/local_lm
+lexicon=data/local/dict/lexiconp.txt
+text_dir=data/rnnlm/text_nosp_${affix}_back
+mkdir -p $dir/config
+set -e
+
+for f in $lexicon; do
+  [ ! -f $f ] && \
+    echo "$0: expected file $f to exist; search for local/wsj_extend_dict.sh in run.sh" && exit 1
+done
+
+#prepare training and dev data
+if [ $stage -le 0 ]; then
+  mkdir -p $text_dir
+  cat $srcdir/train.rnn | awk '{for(i=NF;i>0;i--) printf("%s ",$i); print""}'> $text_dir/chime4.txt
+  sed -i -e "s/<RNN_UNK>/<UNK>/g" $text_dir/chime4.txt
```

Comment:
The -i option to sed is not portable to mac, don't use it.  Anyway I don't see why it's even necessary-- where would RNN_UNK have come from?

Reply:
The RNN_UNK is generated by the LM preparation script in chime4 setting. To make it compatible with RNNLM scripts, we need to change the "RNN_UNK" in train.rnn to "UNK". I'll delete the -i option. Thanks.